django/template/base.py

```diff
@@ -682,22 +714,19 @@ def __init__(self, token, parser):
         self.filters = filters
         self.var = var_obj
 
-    def resolve(self, context, ignore_failures=False):
+    def resolve(self, context):
         if isinstance(self.var, Variable):
             try:
                 obj = self.var.resolve(context)
             except VariableDoesNotExist:
-                if ignore_failures:
-                    obj = None
-                else:
-                    string_if_invalid = context.template.engine.string_if_invalid
-                    if string_if_invalid:
-                        if '%s' in string_if_invalid:
-                            return string_if_invalid % self.var
-                        else:
-                            return string_if_invalid
+                string_if_invalid = context.template.engine.string_if_invalid
+                if string_if_invalid:
+                    if '%s' in string_if_invalid:
+                        obj = UndefinedVariable(string_if_invalid % self.var)
                     else:
-                        obj = string_if_invalid
+                        obj = UndefinedVariable(string_if_invalid)
+                else:
+                    obj = UndefinedVariable(string_if_invalid)
```

Comment:
Might want to pass an explicit empty string here or make `UndefinedVariable.string_rep` default to it and don't pass any arg along.